File: Day1/Examples/Fortran/helloMPI.f90
!
! Hello World MPI Example
!
program helloMPI
  use mpi_f08
  implicit none
  integer :: ierror
  integer :: myRank,uniSize,version,subversion
  character(len=MPI_MAX_PROCESSOR_NAME) :: myName
  integer :: iMyName
  logical :: flag=0

  call MPI_Initialized(flag, ierror)
  if (flag==0) then
    write(6,*) 'MPI_Initialized returned false before MPI_Init.'
  end if

  call MPI_Init(ierror)
  call MPI_Comm_size(MPI_COMM_WORLD,uniSize,ierror)
  call MPI_Comm_rank(MPI_COMM_WORLD,myRank, ierror)
  call MPI_Get_processor_name(myName,iMyName, ierror)
  call MPI_Get_version(version,subversion, ierror)
  write(*,'(a,i0,a,i0,a,a,a,i0,a,i0)')"I am process ", myRank, &
              " out of ", uniSize, " running on ", trim(myName), &
              " with MPI version ", version,".",subversion
  call MPI_Finalize(ierror)

end program helloMPI

File: Day1/Examples/C/helloMPI.c
/*
 * 
 * Hello World MPI Example
 * 
*/
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv){

  int ierror;
  int myRank,uniSize;
  int version, subversion;
  int iMyName;
  char myName[MPI_MAX_PROCESSOR_NAME];

  int flag = 0;
  ierror=MPI_Initialized(&flag);
  if (flag==0){
    printf("MPI_Initialized returned false before MPI_Init.\n");
  }

  ierror=MPI_Init(&argc,&argv);
  ierror=MPI_Comm_size(MPI_COMM_WORLD,&uniSize);
  ierror=MPI_Comm_rank(MPI_COMM_WORLD,&myRank);
  ierror=MPI_Get_processor_name(myName,&iMyName);
  ierror=MPI_Get_version(&version,&subversion);
  printf("I am process %d out of %d running on %s with MPI version %d.%d\n",
      myRank,uniSize,myName,version,subversion);
  ierror=MPI_Finalize();

  return ierror;
}

